flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2310,22 +2314,29 @@ def test_profile_respected(self):
         mandatory profiles.
         """
         for profile in (c.value for c in MandatoryProfiles.iterconstants()):
-            self.api.create_volume_with_profile(profile)
+            self.api.create_volume_with_profile(dataset_id=uuid4(),
```

Comment:
For the reviewer:

Unfortunately when I merged these changes in they were un-used, and unfortunately incomplete. Upon request I can separate maybe 10 lines of this change to a different PR. didn't seem substantial enough for a new PR to me, but at 300 lines this PR I'm sympathetic that this might be hard to digest.